test/time.js

```diff
@@ -3,6 +3,22 @@
 const metatests = require('metatests');
 const common = require('..');
 
+const DefaultDate = Date;
+const dateNowValue = 1561975200000; // 2019-07-01T10:00:00.000Z
+// eslint-disable-next-line no-global-assign
+Date = function(...args) {
+  if (!new.target) {
+    return DefaultDate.apply(DefaultDate, args);
+  }
+  if (args.length !== 0) {
+    return new DefaultDate(...args);
+  }
+  return new DefaultDate(dateNowValue);
+};
+Date.now = function now() {
+  return dateNowValue;
+};
```

Comment:
I assume this is needed to check `if (!date) date = new Date();` branch in those functions but can't we just generate the correct date on the fly?
i.e. 
```js
`${date.getUTCFullYear()}-${(date.getUTCMonth() + 1).toString().padStart(2, '0')}-${date.getUTCDate().toString().padStart(2, '0')}`
```
I know that this is code duplication of what's already present in `nowDate` and `nowDateTime` but I'd prefer that to global state modification.

Reply:
That will possibly be a flaky test, and apart from that, it's the same code used to generate the tested value, which I don't think is a valid way to test it.